lib/color_spinor_pack.cu

```diff
@@ -33,41 +31,57 @@
    ensure that we can always fit within a single thread block.  It is
    this constraint that gives rise for the need to cap the limit for
    block-float support, e.g., max_block_float_nc.
-
-   At present we launch a volume of threads (actually multiples
-   thereof for direction / dimension) and thus we have coalesced reads
-   but not coalesced writes.  A more optimal implementation will
-   launch a surface of threads for each halo giving coalesced writes.
  */
 
 namespace quda {
 
-  template <typename Float, bool block_float, int Ns, int Ms, int Nc, int Mc, typename Arg>
-  class GenericPackGhostLauncher : public TunableVectorYZ {
-    Arg &arg;
-    const ColorSpinorField &meta;
-    unsigned int minThreads() const { return arg.volumeCB; }
-    bool tuneGridDim() const { return false; }
-    bool tuneAuxDim() const { return true; }
-
-  public:
-    inline GenericPackGhostLauncher(Arg &arg, const ColorSpinorField &meta, MemoryLocation *destination)
-      : TunableVectorYZ((Ns/Ms)*(Nc/Mc), 2*arg.nParity), arg(arg), meta(meta) {
-
-      if (meta.Location() == QUDA_CUDA_FIELD_LOCATION) {
-#ifdef JITIFY
-        create_jitify_program("kernels/color_spinor_pack.cuh");
-#endif
+  // this is the maximum number of colors for which we support block-float format
+  constexpr int max_block_float_nc = 96;
+
+  template <typename store_t, typename ghost_store_t, QudaFieldOrder order, int nSpin, int nColor>
+  class PackGhost_ : public TunableKernel3D {
```

Comment:
`PackGhost`? As far as I can tell the use of an underscore at the end of the class name doesn't follow the pattern in other files, but maybe there's a particular reason for this deviation.

Reply:
70cee49a2aa19dce5f7355bc6b6cd46512a9e854